### DISCUSSION TITLE
Brace expansion support?

### DISCUSSION THREAD

**Discussion by eggbean** [chrisant996/clink#489] [Q&A] [answered]
Clink does not appear to support brace expansion.

eg.

    cp foo.{jpg,bak}

    wget https://website.com/bar.{1..10}.jpg

Is it possible to have this work, but my current configuration needs modifying? Or if clink doesn't support this, why is that? It seems to be something that it could support, judging by other such expansions that it does support.

**Comment by chrisant996** [comment, marked as answer]
[Brace expansion](https://www.gnu.org/software/bash/manual/html_node/Brace-Expansion.html) is a [shell expansion](https://www.gnu.org/software/bash/manual/html_node/Tilde-Expansion.html), part of the bash shelf scripting language, not part of the [Readline](https://tiswww.case.edu/php/chet/readline/readline.html) input editor.  Brace expansion can be used in shell scripts, which means it's part of the language.

Tilde expansion makes that confusing, though, since there are two different kinds:  [tilde expansion during completion](https://tiswww.case.edu/php/chet/readline/readline.html#index-expand_002dtilde) which is done by Readline and also [tilde expansion in the shell language](https://www.gnu.org/software/bash/manual/html_node/Tilde-Expansion.html) which is done in the script engine and can be used in scripts (and has slightly different syntax!).  Clink only has the first kind of tilde expansion, which happens in the input editor.

One could write a Lua script to perform brace expansion.  One would have to implement it from scratch, so it might not end up with exactly identical behavior to the bash shell expansion form of brace expansion.
